glide-core/src/client/standalone_client.rs

```diff
@@ -116,10 +125,19 @@ impl StandaloneClient {
             match result {
                 Ok((connection, replication_status)) => {
                     nodes.push(connection);
-                    if primary_index.is_none()
-                        && redis::from_owned_redis_value::<String>(replication_status)
-                            .is_ok_and(|val| val.contains("role:master"))
+                    if redis::from_owned_redis_value::<String>(replication_status)
+                        .is_ok_and(|val| val.contains("role:master"))
                     {
+                        if primary_index.is_some() {
+                            // More than one primary found
+                            return Err(StandaloneClientConnectionError::PrimaryConflictFound(
+                                format!(
+                                    "Primary nodes: {:?}, {:?}",
+                                    nodes.pop(),
+                                    nodes.get(primary_index.unwrap())
+                                ),
+                            ));
+                        }
                         primary_index = Some(nodes.len() - 1);
```

Comment:
(an old code... but): avoid doing match operations that might `panic!` (in case of an overflow).
Use `primary_index = Some(nodes.len().saturating_sub(1));`